src/App.tsx

```diff
@@ -15,49 +16,89 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortField {
+  LENGTH = 'length',
+  ALPHABETICALLY = 'alphabet',
+}
+
+function getPrepared(goods: string[], sortField: string, isReversed: boolean) {
+  let preparedGoods = [...goods];
+
+  if (sortField) {
+    switch (sortField) {
+      case SortField.LENGTH:
+        preparedGoods
+          .sort((good1: string, good2: string) => good2.length - good1.length);
+        break;
+      case SortField.ALPHABETICALLY:
+        preparedGoods
+          .sort((good1: string, good2: string) => good1.localeCompare(good2));
+        break;
+      default:
+        break;
+    }
+  }
+
+  if (isReversed) {
+    preparedGoods = preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReversed, SetIsReversed] = useState(false);
+
+  const visibleGoods = getPrepared(goodsFromServer, sortField, isReversed);
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={`button is-info ${sortField === SortField.ALPHABETICALLY ? '' : 'is-light'}`}
+          onClick={() => {
+            setSortField(SortField.ALPHABETICALLY);
+          }}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={`button is-success ${sortField === SortField.LENGTH ? '' : 'is-light'}`}
+          onClick={() => {
+            setSortField(SortField.LENGTH);
+          }}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={`button is-warning ${isReversed ? '' : 'is-light'}`}
+          onClick={() => {
+            SetIsReversed(!isReversed);
+            getPrepared(goodsFromServer, sortField, isReversed);
+          }}
         >
           Reverse
         </button>
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {sortField !== '' ? (
```

Comment:
<img width="594" alt="image" src="https://github.com/mate-academy/react_list-of-goods-hooks/assets/70220667/0ce22713-3bb0-44a3-98e1-8013a94a94fc">
you also should show the reset button in this case